plugins/state_history_plugin/include/eosio/state_history_plugin/session.hpp

```diff
@@ -447,7 +447,7 @@ struct session : session_base, std::enable_shared_from_this<session<Plugin, Sock
 
    void update_current_request(state_history::get_blocks_request_v0& req) {
       fc_dlog(plugin->logger(), "replying get_blocks_request_v0 = ${req}", ("req", req));
-      to_send_block_num = req.start_block_num;
+      to_send_block_num = std::max(req.start_block_num, plugin->get_first_available_block_num());
```

Comment:
not related to this change, but `plugin` is such a bad name for this member... I'd call it `ship` or something.